pandas/tools/plotting.py

```diff
@@ -1458,22 +1464,33 @@ def match_labels(data, e):
 
         elif com.is_list_like(err):
             if com.is_iterator(err):
-                err = np.atleast_2d(list(err))
+                err = np.asanyarray(list(err))
             else:
                 # raw error values
-                err = np.atleast_2d(err)
+                err = np.asanyarray(err)
 
-            err_shape = err.shape
+            if self.nseries == 1 and err.ndim == 2 and len(err) == 2:
```

Comment:
I would like to see a more general soln here, rather than specific ndim checking.